src/txmempool.cpp

```diff
@@ -1197,3 +1216,58 @@ void CTxMemPool::SetLoadTried(bool load_tried)
     LOCK(cs);
     m_load_tried = load_tried;
 }
+
+std::vector<CTxMemPool::txiter> CTxMemPool::CalculateCluster(const std::vector<uint256>& txids) const
+{
+    AssertLockHeld(cs);
+    std::vector<txiter> cluster{GetIterVec(txids)};
+    assert(cluster.size() == txids.size());
+    cluster.reserve(std::accumulate(cluster.cbegin(), cluster.cend(), 0, [](size_t sum, const auto it) {
+        return sum + it->GetCountWithAncestors() + it->GetCountWithDescendants(); }));
+    {
+        // Use epoch: visiting an entry means we have added it to the cluster vector.
+        WITH_FRESH_EPOCH(m_epoch);
+        for (const auto& it : cluster) {
+            visited(it);
+        }
+        // i = index of where the list of unprocessed starts
+        for (size_t i{0}, unprocessed_count{txids.size()}; i < unprocessed_count; ++i) {
+            const auto curr = cluster[i];
+            for (const CTxMemPoolEntry& parent_entry : curr->GetMemPoolParentsConst()) {
+                const auto parent_it = mapTx.iterator_to(parent_entry);
+                if (!visited(parent_it)) {
+                    if (parent_entry.GetCountWithAncestors() == 1 &&
+                        parent_entry.GetCountWithDescendants() == 1) {
+                        // We don't need to process this entry. Just add it to the processed
+                        // transactions and skip.
+                        std::swap(cluster[i+1], cluster.back());
+                        cluster[i+1] = parent_it;
+                        ++i; 
```

Comment:
Sorry, lint error is my fault here

```suggestion
                        ++i;
```